src/API/TransactionName.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Solarwinds\ApmPhp\API;
+
+use OpenTelemetry\API\Behavior\LogsMessagesTrait;
+use OpenTelemetry\API\Trace\LocalRootSpan;
+use OpenTelemetry\SDK\Trace\ReadableSpanInterface;
+use Solarwinds\ApmPhp\Trace\SpanProcessor\TransactionNameSpanProcessor;
+
+final class TransactionName
+{
+    use LogsMessagesTrait;
+
+    /**
+     * Retrieve the local root span (root-most active span which has a remote or invalid parent) and try to set transaction name in it
+     */
+    public static function set(string $name): bool
```

Comment:
Could this function also account for these cases?:

1. `$name` is empty (is that allowed in php?)
2. Agent is not enabled (if that logic is implemented)

Reply:
1. if `$name` is empty, it will set an empty string to `sw.transaction` to respect user input.
2. If agent is disabled, `span = LocalRootSpan::current();` will return an invalid span that the function will return false. Nothing set.

Reply:
Empty name should be considered invalid. Take a look at [our spec](https://swicloud.atlassian.net/wiki/spaces/NIT/pages/3601400601/Swo-SDK+Application+Programming+Interface) regarding custom transaction name API.

Reply:
I see. Updated not allowing empty string